step-release-vis/src/app/components/environments/environments_test.ts

```diff
@@ -51,6 +56,31 @@ describe('EnvironmentsComponent', () => {
   });
 
   it('candidates should be added to service', () => {
-    // TODO(#176): add test
+    const candNames = getCandNames();
+    expect(candNames.size).toEqual(candidateServiceStub.candColors.size);
+    candNames.forEach(candName => {
+      expect(candidateServiceStub.candColors.has(candName)).toBeTrue();
+    });
+  });
+
+  it('candidates should have unique colors', () => {
+    const candColors = new Set<number>();
+    getCandNames().forEach(candName => {
+      const candColor = candidateServiceStub.candColors.get(candName);
+      expect(candColor).toBeTruthy();
+      expect(candColors.has(candColor)).toBeFalse();
```

Comment:
Is this expect needed? candColors seems to be an empty set all the time.

Reply:
Forgot to update the set. Fixed it now.

Reply:
You should randomly backtest your tests with input which would fail the test. This way you can make sure your tests are not producing false positive result, but actually test something :)

Reply:
I am not saying you should do that for every test, that would be really expensive, but you should do it regularly with random tests.